### PR TITLE
Unit test shaders

### PR DESCRIPTION
**Pull request by jamesmporter**
Addresses #1382

### CODE REVIEW

File: packages/d3fc-webgl/test/util/vertextShaderUtil.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export const getShaders = element => {
+    let _program;
+    try {
+        const bar = element.decorate(program => {
+            _program = program;
+            throw new Error();
+        });
+        bar();
+    } catch (e) {
+        // no-op
```
Comment (chrisprice) [inline packages/d3fc-webgl/test/util/vertextShaderUtil.js]: Might be worth throwing a known error message i.e. `new Error('HORRIBLE HACK')` and asserting that's the received error i.e. `catch (e) { if (e.message !== 'HORRIBLE HACK') { throw e; } ... }`

File: packages/d3fc-webgl/package.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
   },
   "dependencies": {
     "@d3fc/d3fc-rebind": "^5.0.8",
-    "d3": "^4.1.1"
+    "d3": "^4.1.1",
+    "glsl-transpiler": "^1.8.5"
```
Comment (chrisprice) [inline packages/d3fc-webgl/package.json]: As this is only required for testing, please can you make it a devDependency?

**Review by chrisprice — CHANGES_REQUESTED** [review]
Very nice. I think this will make a big difference to the maintainability of these shaders. Just a few minor comments

File: packages/d3fc-webgl/package.json (hunk 2 above)
Comment (jamesmporter) [reply packages/d3fc-webgl/package.json]: Sure.

File: packages/d3fc-webgl/test/util/vertextShaderUtil.js (hunk 1 above)
Comment (jamesmporter) [reply packages/d3fc-webgl/test/util/vertextShaderUtil.js]: Good idea.

**Review by chrisprice — APPROVED** [review]

**Review by OliverForeman — APPROVED** [review]

**Pull request merged**